src/main/java/bowling/domain/Frames.java

```diff
@@ -74,18 +75,22 @@ public int totalScore() {
                 .sum();
     }
 
-    @Override
-    public Iterator<Frame> iterator() {
-        return frames.iterator();
+    public FrameBoard assembleFrameBoard(Participant participant) {
+        return new FrameBoard(participant.getName(), nextTurnNumber(), assembleFrameInfos(), isFinished());
     }
 
-    @Override
-    public void forEach(Consumer<? super Frame> action) {
-        Iterable.super.forEach(action);
+    private List<FrameInfo> assembleFrameInfos() {
+        List<FrameInfo> frameInfos = new ArrayList<>();
+        for (Frame frame : frames) {
+            FrameInfo frameInfo = frame.assembleFrameInfo();
+            frameInfos.add(frameInfo);
+        }
+        return frameInfos;
```

Comment:
```suggestion
        return frames.stream()
                .map(Frame::assembleFrameInfo)
                .collect(Collectors.toList());
```
nit : 이제보니 여기도 stream을 사용해도 되겠어요.